### PR TITLE
Reonucam

### PR DESCRIPTION
**Pull request by Reonu**

### CODE REVIEW

File: src/game/game_init.h [hunk 1]
```diff
@@ -30,7 +30,6 @@ enum ZBmodes {
     KEEP_ZBUFFER = 0,
     CLEAR_ZBUFFER = 1,
 };
-
```
Comment (thecozies) [inline src/game/game_init.h:33]: >:(

Comment (Reonu) [reply src/game/game_init.h:33]: Sorry @Arceveti broke into my house and did that

File: src/game/camera.c [hunk 2]
```diff
@@ -1115,13 +1159,60 @@ s32 snap_to_45_degrees(s16 angle) {
...
+            set_cam_angle(CAM_ANGLE_MARIO); // Enter mario cam if R is pressed 2 times in less than 5 frames
+            gReonucamState.rButtonCounter2 = 6;
+        } else {
+            gReonucamState.rButtonCounter2 = 0;
+        }
+    } else {
+        gReonucamState.rButtonCounter2++;
+     }
+    if (gPlayer1Controller->buttonPressed & D_JPAD) {
+       s8DirModeBaseYaw = (s8DirModeBaseYaw + 0x1000) & 0xE000; // Lock the camera to the nearest 45deg axis
```
Comment (thecozies) [inline src/game/camera.c:1201]: this function always works! ive had issues with the other method at certain angles

```suggestion
       s8DirModeBaseYaw = snap_to_45_degrees(s8DirModeBaseYaw); // Lock the camera to the nearest 45deg axis
```

File: src/game/camera.c [hunk 3]
```diff
@@ -2875,6 +2974,11 @@ void update_camera(struct Camera *c) {
     OSTime first   = osGetTime();
     OSTime colTime = collisionTime[perfIteration];
 #endif
+
+#ifdef REONUCAM
+    extern s16 s8DirModeBaseYaw;
+#endif
```
Comment (thecozies) [inline src/game/camera.c:2980]: are you sure this is needed? i thought this was already declared at the top

File: src/game/camera.c [hunk 4]
```diff
@@ -4551,15 +4664,21 @@ void play_camera_buzz_if_c_sideways(void) {
...
 }
 
 void play_sound_cbutton_down(void) {
+#ifndef REONUCAM
     play_sound(SOUND_MENU_CAMERA_ZOOM_OUT, gGlobalSoundSource);
+#endif
 }
 
 void play_sound_cbutton_side(void) {
+#ifndef REONUCAM
```
Comment (thecozies) [inline src/game/camera.c:4679]: when we pull in reonucam to 2.1 we need to make these a `DISABLE_CAMERA_TURN_SOUNDS`

File: src/game/camera.c (hunk 3 above)
Comment (Reonu) [reply src/game/camera.c:2980]: lmao yeah that makes no sense

File: src/game/camera.c (hunk 4 above)
Comment (Reonu) [reply src/game/camera.c:4675]: sure!

**Pull request closed without merging**